clients/include/hipsparse_test_unique_ptr.hpp

```diff
@@ -257,6 +257,22 @@ namespace hipsparse_test
         }
     };
 
+    struct spgemm_struct
```

Comment:
why do we add this layer?

Reply:
temporary storage buffer is not propagated to spgemm_copy function, but we need it for rocsparse. We store the pointer there. Users could deallocate it before the call to spgemm_copy but I do not see any other way to stay compatible with cusparse.